common/plugins/eu.esdihumboldt.hale.common.headless/src/eu/esdihumboldt/hale/common/headless/transform/filter/InstanceFilterDefinition.java

```diff
@@ -183,20 +186,25 @@ private boolean applyTypedFilter(Instance instance) {
 				return false;
 			}
 		}
-		// it reaches here that means Instance does not match any type of type
-		// filters.
-		// Method should return true.
+		// it reaches here that means Instance does not match any type of
+		// type filters. Method should return true.
 		return true;
 	}
 
 	private boolean applyExcludeFilters(Instance instance) {
 		// for excluded Types
 		for (String excludedType : this.excludedTypes)
-			if (instance.getDefinition().getName().getLocalPart().equals(excludedType + "Type"))
-				return false;
+			if (checkType(instance.getDefinition(), excludedType))
+				return true;
+		// instance does not match any excluded types.
+		return false;
+	}
 
-		// instance does not match any excludeed types.
-		return true;
+	// Checking type of instance with givenType.
+	private boolean checkType(TypeDefinition instanceType, String givenType) {
+		// Here giveType is supplied by user which should be trailed by "Type"
+		givenType += "Type";
```

Comment:
There should be different ways to compare the type, based on the given type expression:

1. Check for equality with the fully qualified type name (including the namespace information)
2. If no namespace is provided in the given expression, check for equality with the type's display name
3. Check for equality with the fully qualified XML element names (including the namespace information)